### ISSUE TITLE
grpcwebproxy crashing 

### ISSUE THREAD

**Issue by rmilejcz** [improbable-eng/grpc-web#971]
**Versions of relevant software used**
Latest

**What happened**
The proxy stops working randomly, and I am not sure why. I use this startup config:
```
"/grpcwebproxy --allowed_origins=https://app.mywebsite.com --server_bind_address=0.0.0.0 --run_tls_server --run_http_server=false --use_websockets --server_tls_cert_file=/certs/core.kalosflorida.com --server_tls_key_file=/certs/core.kalosflorida.com --backend_addr=rpc:8418 --backend_max_call_recv_msg_size=31457280 --server_http_max_write_timeout=600s --server_http_max_read_timeout=600s"
```

The only debug information I get is that the call was cancelled, and I can still hit and get responses from the RPC server via grpc-gateway, the proxy is the only thing that seems to be breaking.

**What you expected to happen**

**How to reproduce it (as minimally and precisely as possible)**:
I regretfully can't provide this, I can barely reproduce this myself. I just wait for it to occur in production and try and suss out what caused it, but at this point I am stumped.

**Full logs to relevant components**
```
proxy_1        | time="2021-05-06T17:54:17Z" level=info msg="finished streaming call with code Canceled" error="rpc error: code = Canceled desc = context canceled" grpc.code=Canceled grpc.method=BatchGet grpc.service=PendingBillingService grpc.start_time="2021-05-06T17:54:05Z" grpc.time_ms=12218.064 span.kind=server system=grpc
```

Basically every single RPC request that comes up will start to return this message, with code Canceled. I have no explicit calls to cancel in my own code, so I'm not sure what's calling it.

**Anything else we need to know**

I have been using this binary in production for a long time and have not had any issues. I do not know if the issue is actually within grpcwebproxy but I am hoping someone here can at least point me in a direction that helps me figure out what is going on. I know this is sparse and I apologize, I am mainly looking for some help debugging this issue and secretly hoping the solution is obvious and in front of my face

**Comment by rmilejcz** [comment]
I just noticed this log has a slightly different error message: ` failed proxying s2c`
```
proxy_1        | time="2021-05-06T18:25:08Z" level=error msg="finished streaming call with code Internal" error="rpc error: code = Internal desc = failed proxying s2c: rpc error: code = Canceled desc = context canceled" grpc.code=Internal grpc.method=BatchGet grpc.service=DocumentService grpc.start_time="2021-05-06T18:25:08Z" grpc.time_ms=4.612 span.kind=server system=grpc
```

Not sure what that means, will do some searching hope it helps

**Comment by rmilejcz** [comment]
I was able to rule out `grpcwebproxy` as the cause of the issue, so I will close this now, but if anyone has any suggestions I could implement to get a better picture of what is going on under the hood it would be much appreciated

**Comment by johanbrandhorst** [comment]
Hi Robbie,

It sounds to me like what you're seeing is the effects of users closing the browser session. It will manifest as a canceled context in the backend. Could that be the culprit? Other sources of context cancelation could be a proxy somewhere reaching a request timeout.

**Comment by johanbrandhorst** [comment]
For example, reloading the page or closing the browser can cause it.
